documentation/external_ssl_termination.markdown

```diff
@@ -5,21 +5,61 @@ canonical: "/puppetserver/latest/external_ssl_termination.html"
+>
+> With `allow-header-cert-info` set to 'true', core Ruby Puppet application code will use only the client HTTP header values---not an SSL-layer client certificate---to determine the client subject name, authentication status, and trusted facts. This is true even if the web server is hosting an HTTPS connection. This applies to validation of the client via rules in the [auth.conf](https://docs.puppetlabs.com/guides/rest_auth_conf.html) file and any [trusted facts](https://docs.puppetlabs.com/puppet/latest/reference/lang_facts_and_builtin_vars.html#trusted-facts) extracted from certificate extensions.
+>
+> If the `client-auth` setting in the `webserver` config block is set to `need` or `want`, the Jetty web server will still validate the client certificate against a certificate authority store, but it will only verify the SSL-layer client certificate---not a certificate in an  `X-Client-Cert` header.
+
+
+## Restart Puppet Server
+
+You'll need to restart Puppet Server for the configuration changes to take effect.
+
+## Configure SSL Terminating Proxy to Set HTTP Headers
+
+The device that terminates SSL for Puppet Server must extract information from the client's certificate and insert that information into three HTTP headers. See the documentation for your SSL terminator for details.
+
+The headers you'll need to set are `X-Client-Verify`, `X-Client-DN`, and `X-Client-Cert`.
+
+### `X-Client-Verify`
+
+Mandatory. Must be one of `NONE`, `SUCCESS`, `GENEROUS` or `FAILED:reason`. Puppet Server uses this to authorize requests.
+
+You can change this header name with [the `ssl_client_verify_header` setting.](https://docs.puppetlabs.com/references/latest/configuration.html#sslclientverifyheader)
+
+This setting (and its twin, `ssl_client_header`) is a bit odd: its value should be the result of transforming the desired HTTP header name into a CGI-style environment variable name. That is, to change the HTTP header to `X-SSL-Client-Verify`, you would set the setting to `HTTP_X_SSL_CLIENT_VERIFY`. (Add `HTTP_` to the front, change hyphens to underscores, and uppercase everything.)
+
+(Puppet Server will eventually UN-munge the CGI variable name to get a valid HTTP header name, and use that name to interact directly with an HTTP request. This is a legacy quirk to ensure that the setting works the same for both Puppet Server and a Rack Puppet master; note that Rack actually does use CGI environment variables.)
+
+### `X-Client-DN`
+
+Mandatory. Must be the [Subject DN][] of the agent's certificate, if a certificate was presented. Puppet Server uses this to authorize requests.
+
+You can change this header name with [the `ssl_client_header` setting.](https://docs.puppetlabs.com/references/latest/configuration.html#sslclientheader) See the note above for more info about this setting's expected values.
```

Comment:
We filed a bug around this recently, SERVER-213, which talks about Puppet Server's inability to decode the `X-Client-DN` when the client DN is in "OpenSSL" format - using slash-delimiters - as opposed to RFC-2253 - using comma-delimiters.  Might make sense to document that here so that others are aware of the outstanding problem?